youtube_dl/YoutubeDL.py

```diff
@@ -1291,6 +1289,22 @@ def process_info(self, info_dict):
             self.report_error('unable to create directory ' + compat_str(err))
             return
 
+        filesystem_enc = sys.getfilesystemencoding()
+        if filesystem_enc is None:
+            filesystem_enc = 'utf-8'
+        max_filename_len = 200
+        if sys.platform.startswith('linux'):
```

Comment:
what extactly the thing i need to check with hasattr.
both sys.getfilesystemencoding and sys.platform are available in python 2 and 3:
https://docs.python.org/2/library/sys.html#sys.getfilesystemencoding
https://docs.python.org/3/library/sys.html#sys.getfilesystemencoding
https://docs.python.org/2/library/sys.html#sys.platform
https://docs.python.org/3/library/sys.html#sys.platform